### PR TITLE
Bug 1508227 - Provide a better example for using Glean

### PR DESCRIPTION
**Pull request by travis79**
Added GleanApplication which extends Application and moved the Glean initialization from the Activity onCreate to the application onCreate.

### CODE REVIEW

File: samples/glean/src/main/java/org/mozilla/samples/glean/GleanApplication.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        // must be done right after enabling logging.
+        Glean.initialize(applicationContext, Configuration())
+
+        // Set a sample value for a metric.
+        GleanMetrics.Basic.os.set("Android")
+    }
+
+    companion object {
+        const val NON_FATAL_CRASH_BROADCAST = "org.mozilla.samples.glean"
+    }
```
Comment (pocmo) [inline samples/glean/src/main/java/org/mozilla/samples/glean/GleanApplication.kt]: This companion object and the constant should not be needed here.

**Comment by travis79** [comment]
Requested changes have been made.

**Comment by pocmo** [comment]
Please squash that PR before merging. :)

**Comment by travis79** [comment]
I believe I have squashed it (correctly) now.  Had to expand my git-fu a bit but I think I got it straightened out now.

**Pull request merged**